dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/passes/reachingdef/DdgGenerator.scala

```diff
@@ -172,10 +172,27 @@ class DdgGenerator(semantics: Semantics) {
       val identifierDestPairs =
         method._identifierViaContainsOut
           .flatMap { identifier =>
-            identifierToFirstUsages(identifier).map(usage => (identifier, usage))
+            val firstAndLastUsageByMethod = identifierToFirstUsages(identifier).groupBy(_.method)
+            firstAndLastUsageByMethod.values.flatMap { usage =>
+              if (usage.nonEmpty) {
+                val firstUsage = usage.head // First usage of identifier
+                val lastUsage  = usage.last // Last usage of identifier
+                if (
+                  identifier.lineNumber.isDefined && firstUsage.lineNumber.isDefined && (identifier.lineNumber.get <= firstUsage.lineNumber.get)
+                )
+                  Some(identifier, firstUsage)
+                else if (
+                  identifier.lineNumber.isDefined && lastUsage.lineNumber.isDefined && (identifier.lineNumber.get >= lastUsage.lineNumber.get)
+                )
+                  Some(lastUsage, identifier)
```

Comment:
Why is the order swapped here? Should it not be `Some(identifier, lastUsage)`?

Reply:
having the edge from `lastUsage` to `identifier` empowers us to get the logical flow, as the final flow should be from a linenumber which is less than the node with a higher linenumber

Reply:
Ah, right now I understand, sorry seems I had a misunderstanding of the output